Objects/cown.c

```diff
@@ -156,6 +160,13 @@ static PyObject *PyCown_release(PyCownObject *self, PyObject *ignored) {
     BAIL_UNLESS_OWNED(self, "Thread attempted to release a cown it did not own");
 
     Py_BEGIN_ALLOW_THREADS
+    if (self->value && Py_TYPE(self->value) == &PyRegion_Type) {
+        if (PyCown_close_region(self->value) == NULL) {
+            // Close region failed -- propagate its error
+            return NULL;
```

Comment:
I believe this has to require the GIL again, right? Or maybe move the `Py_BEGIN_ALLOW_THREADS` after the the if statement?

```suggestion
            Py_END_ALLOW_THREADS
            return NULL;
```

Reply:
This is a good catch but your fix is incorrrect because the macro contains a block end. I'll fix this.

Reply:
After thinking some more I realise we don't need to release the GIL here. There should never be two threads racing on releasing the same cown (and if that happens because of buggy code, we don't care if that race is serialised by the GIL). Acquring needs to release the GIL because it currently blocks the underlying Python thread until acquisition and we don't want to hold the GIL while blocking. So this is resolved by removing the macros from release but keeping them in acquire.